sign/bls/bls_test.go

```diff
@@ -6,23 +6,49 @@ import (
 
 	"github.com/stretchr/testify/require"
 	"go.dedis.ch/kyber/v3"
+	"go.dedis.ch/kyber/v3/pairing"
+	"go.dedis.ch/kyber/v3/pairing/bls12381"
 	"go.dedis.ch/kyber/v3/pairing/bn256"
 	"go.dedis.ch/kyber/v3/util/random"
 )
 
-func TestBLS(t *testing.T) {
-	msg := []byte("Hello Boneh-Lynn-Shacham")
+func TestBn256(t *testing.T) {
 	suite := bn256.NewSuite()
+	testBLS(t, suite)
+	testBLSFailSig(t, suite)
+	testBLSFailKey(t, suite)
+	testBLSAggregateSignatures(t, suite)
+	testBLSFailAggregatedSig(t, suite)
+	testBLSFailAggregatedKey(t, suite)
+	testBLSBatchVerify(t, suite)
+	testBLSFailBatchVerify(t, suite)
+	testBinaryMarshalAfterAggregationIssue400(t, suite)
+}
+
+func TestBLS12381(t *testing.T) {
+	suite := bls12381.NewSuite()
+	testBLS(t, suite)
+	testBLSFailSig(t, suite)
+	testBLSFailKey(t, suite)
+	testBLSAggregateSignatures(t, suite)
```

Comment:
Thank you very much for your interest and contribution.

However, there is a security issue: BLS12-381 is also vulnerable to rogue public-key attacks. You shouldn't use AggregateSignatures() and AggregatePublicKeys() from sign/bls as you are currently testing here.

Instead, you should move these testBLSAggregateSignatures(), testBLSFailAggregatedSig() and testBLSFailAggregatedKey() to sign/bdn